src/units/unit.hpp

```diff
@@ -1754,10 +1754,12 @@ class unit : public std::enable_shared_from_this<unit>
 	bool has_ability_by_id(const std::string& ability) const;
 
 	/**
-	 * Removes a unit's abilities with a specific ID.
-	 * @param ability The type of ability (tag name) to remove.
+	 * Removes a unit's abilities with a specific ID or other attribute.
+	 * @param filter the config of ability to remove.
+	 * @param ability The type of ability (tag) to remove.
+	 * @param in_tag if attribute checked was in tag abilities or in filter tag
 	 */
-	void remove_ability_by_id(const std::string& ability);
+	void remove_ability_by_attribute(const config& filter, const std::string& ability = "", bool in_tag = false);
```

Comment:
If a function is only called from two places, then I think it's clearer have no default values for arguments. With only two calling locations, just put the argument values in the calling location.

Reply:
done